### DISCUSSION TITLE
import cv2 problem

### DISCUSSION THREAD

**Discussion by Matin-GorganSaraj** [alexisInSAR/EZ-InSAR#40] [Help]
hello 
when i run the pre-run of isce processing step i face with this problem ( not model named cv2 [i think it's opencv])
![Screenshot (183)](https://user-images.githubusercontent.com/130948662/235311141-65e76de3-88e1-4523-a06f-321fa5e84172.png)
when i check the code i see cv2 don't used in Stack.py then i delete the import cv2 line in Stack.py (line 1646) then this step run well ! 
![Screenshot (184)](https://user-images.githubusercontent.com/130948662/235311260-679a1126-7623-403f-ba4b-1faa5585e454.png)
![Screenshot (185)](https://user-images.githubusercontent.com/130948662/235311269-b15e72fa-40cd-4eae-aee0-fbcf777c2e62.png)
and my question is , the changing into Stack.py can cause me any problem ? 
With all regard
Matin

**Comment by alexisInSAR** [comment]
I think the best solution is just to install cv2: 
````
conda install opencv
````
or 
````
pip3 install opencv-python
````

**Comment by Matin-GorganSaraj** [comment]
yes i install opencv and it runs well , thank you for responding
but i have another quastion ; can i proccess with my gpu in isce2 , i think , i saw something on isce docu but i dont know how to do it.

**Reply by alexisInSAR** [reply]
Currently, the installation of ISCE is done via "conda", so I don't think the CUDA environment will be set up during this installation step. However, here are some potential tips on how to use the GPU processors (I have never used it, so I hope I am not making mistakes). 

Firstly, EZ-InSAR asks to install ISCE in "two" steps: 
1. "conda install" to get the compiled and executable scripts (core of the InSAR processor); 
2. "git clone" to download the other scripts used by EZ-InSAR (links between EZ-InSAR and ISCE). 

In this case, I think you can avoid installing the compiled and executable scripts using "conda install": 
1. You need to check hardware compatibility: CUDA environment, NVIDIA GPU, etc. I think GPU processing is powerful if you have [a suitable GPU card](https://developer.nvidia.com/cuda-gpus).
2. You need to install the CUDA environment in the appropriate version regarding your hardware. 
3. You must uninstall the conda installation using "conda uninstall isce2". 
4. You should now be able to compile the ISCE scripts from the downloaded files with the "git clone" command. Please see the ISCE documentation [here](https://github.com/isce-framework/isce2). 

Unfortunately, I don't know if the CUDA compiled scripts are used by default. So you can test, and if no, I will update EZ-InSAR to add a new property in the ISCE input cards.
